### PR TITLE
161 remove tnc accessegress skims from the model

### PR DESCRIPTION
**Pull request by aletzdy**
## Proposed changes

Removing TNC access/egress skims from the model

## Impact
This update will reduce model runtime and disk usage:

- Disk space reduced by about 54gb
- OMX skims: 10gb
- Emme transit database: 44gb
- ActivitySim skim buffer memory usage reduced by about 35gb
- Total runtime reduced by about 2 hours 20 minutes
    - Transit skimming: 9 min per iteration
    - Resident model mp_setup_skims: 5 min per iteration,
    - Additional resident model savings: 10-15 min per iteration (can't identify where, not visible in timing_log.csv)
    - Copy to T drive: 50 min

## Types of changes

What types of changes does your code introduce to ABM?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] Model was run from to start to finish without error.
- [x] Model results were examined. VMT and ridership stayed similar, with only small differences.
## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Further comments

Code was reviewed by SANDAG staff.

### CODE REVIEW

**Comment by aber-sandag** [comment]
Visitor model trip mode choice preprocessor still references TNC skims. However, these lines (122-134) are no longer used, so it would be best to just remove them. Does not need to be part of this PR unless these lines result in errors with missing TNC skims.

https://github.com/SANDAG/ABM/blob/161-remove-tnc-accessegress-skims-from-the-model/src/asim/configs/visitor/trip_mode_choice_annotate_trips_preprocessor.csv#L122-L134

File: src/asim/configs/resident/trip_mode_choice.csv [hunk 1]
```diff
@@ -292,102 +292,102 @@ util_KNR_MIX - Destination Employment Density,KNR_MIX - Destination Employment D
...
-util_TNCIN_PRM_wait_access_time_(at_attraction_end),TNCIN_PRM - Access mt/nev wait time (at attraction end),"@np.where(df.nev_prm_access_available_in, nevWaitTime, np.where(df.microtransit_prm_access_available_in, microtransitWaitTime, 0)) * df.time_factor * ~df.outbound",,,,,,,,,,,,,,,,coef_wait,,,,,,,
-util_TNCIN_PRM_Walk_other_time,TNCIN_PRM - Walk other time,@(odt_skims['TNCOUT_PRM_XFERWALK']) * df.time_factor * ~df.outbound,,,,,,,,,,,,,,,,coef_xwalk,,,,,,,
-util_TNCIN_PRM_Fare,TNCIN_PRM - Fare,"@df.transitSubsidyPassDiscount*(odt_skims['TNCIN_PRM_FARE'])*100*df.number_of_participants/(np.maximum(df.income,1000)**df.income_exponent) * ~df.outbound",,,,,,,,,,,,,,,,coef_income,,,,,,,
-util_TNCIN_PRM_ASC,TNCIN_PRM - Alternative-specific constant,@df.tnc_prm_inb_asc * df.time_factor * ~df.outbound,,,,,,,,,,,,,,,,coef_ivt,,,,,,,
+util_KNRIN_PRM_In_vehicle_time,KNRIN_PRM - In-vehicle time,@(odt_skims['KNRIN_PRM_TOTALIVTT']) * df.time_factor * ~df.outbound,,,,,,,,,,,,,,,,coef_ivt,,,,,,,
```
Comment (aber-sandag) [inline src/asim/configs/resident/trip_mode_choice.csv:338]: Labels and descriptions were incorrectly changed here, presumably through find-and-replace. No impact on functionality but could cause confusion/

**Comment by aber-sandag** [comment]
TNC skims still referenced in write trip matrices annotator for special market models. These models do not use TNC-transit mode, so these lines are unused and should be removed. As above, does not need to be part of this PR unless it is causing errors.

https://github.com/SANDAG/ABM/blob/161-remove-tnc-accessegress-skims-from-the-model/src/asim/configs/crossborder/write_trip_matrices_annotate_trips_preprocessor.csv#L299

**Review by aber-sandag — CHANGES_REQUESTED** [review]
See comments above. Incorrect label changes in trip mode choice should be fixed, and ensure unused lines referencing TNC skims do not cause issues.

**Comment by aber-sandag** [comment]
> TNC skims still referenced in write trip matrices annotator for crossborder and visitor models. These models do not use TNC-transit mode, so these lines are unused and should be removed. As above, does not need to be part of this PR unless it is causing errors.
> 
> https://github.com/SANDAG/ABM/blob/161-remove-tnc-accessegress-skims-from-the-model/src/asim/configs/crossborder/write_trip_matrices_annotate_trips_preprocessor.csv#L299

I ran a scenario with these changes, and it crashed in Crossborder due to this issue (AssertionError: 3d skim key TNCOUT_LOC_TOTALIVTT not in skims). Presumably all cases I mentioned previously where TNC skims are still referenced (though unused) need to be resolved. It should be possible to just remove these lines.

**Comment by aber-sandag** [comment]
After removing the mentioned lines, 2022 and 2035 build scenarios completed successfully. Did not find any clear differences in mode share compared to previous runs, though difficult to tell due to low volume of TNC transit trips. 
- Disk space reduced by about 54gb
  - OMX skims: 10gb
  - Emme transit database: 44gb
- ActivitySim skim buffer memory usage reduced by about 35gb
- Total runtime reduced by about 2 hours 20 minutes
  - Transit skimming: 9 min per iteration
  - Resident model mp_setup_skims: 5 min per iteration, 
  - Additional resident model savings: 10-15 min per iteration (can't identify where, not visible in timing_log.csv)
  - Copy to T drive: 50 min

File: src/asim/configs/resident/trip_mode_choice.csv (hunk 1 above)
Comment (aletzdy) [reply src/asim/configs/resident/trip_mode_choice.csv:338]: fixed.

**Review by aber-sandag — APPROVED** [review]
Fixes look good to me

File: src/main/emme/toolbox/initialize/initialize_matrices.py [hunk 2]
```diff
@@ -195,7 +195,7 @@ def transit_demand(self):
             ("MIX",  "all modes xfer pen demand"),
         ]
         for period in self._all_periods:
-            for a_name in ["WALK", "PNROUT", "PNRIN", "KNROUT", "KNRIN", "TNCIN", "TNCOUT"]:
+            for a_name in ["WALK", "PNROUT", "PNRIN", "KNROUT", "KNRIN", "TNCOUT", "TNCIN"]:
```
Comment (bhargavasana) [inline src/main/emme/toolbox/initialize/initialize_matrices.py:198]: @aletzdy should these be deleted here?

Comment (bhargavasana) [reply src/main/emme/toolbox/initialize/initialize_matrices.py:198]: Alexander suggested that these might be still needed for assignment.

**Review by bhargavasana — APPROVED** [review]

**Pull request merged**